src/main/java/edu/ksu/canvas/attendance/submitter/AssignmentValidator.java

```diff
@@ -0,0 +1,96 @@
+package edu.ksu.canvas.attendance.submitter;
+
+
+import edu.ksu.canvas.attendance.entity.AttendanceAssignment;
+import edu.ksu.canvas.attendance.exception.CanvasOutOfSyncException;
+import edu.ksu.canvas.attendance.model.AttendanceSummaryModel;
+import edu.ksu.canvas.attendance.services.CanvasApiWrapperService;
+import edu.ksu.canvas.model.assignment.Assignment;
+import edu.ksu.canvas.oauth.OauthToken;
+import org.apache.log4j.Logger;
+import org.springframework.stereotype.Component;
+
+import java.io.IOException;
+import java.util.Optional;
+
+@Component
+public class AssignmentValidator {
+
+    private static final Logger LOG = Logger.getLogger(AssignmentValidator.class);
+
+    public void validateAttendanceAssignment(Long courseId, AttendanceAssignment attendanceAssignment, CanvasApiWrapperService canvasApiWrapperService, OauthToken oauthToken) throws IOException, CanvasOutOfSyncException {
+
+        // If there is no linked canvas assignment, creates on in canvas
+        if(attendanceAssignment.getCanvasAssignmentId() == null) {
+            LOG.info("No Canvas assignment linked to section: " + attendanceAssignment.getAttendanceSection().getCanvasSectionId());
+            throw new CanvasOutOfSyncException("NO CANVAS ASSIGNMENT LINKED");
+        }
+
+        // Looks for the linked assignment in canvas
+        Optional<Assignment> assignmentOptional;
+        try {
+            assignmentOptional = canvasApiWrapperService.getSingleAssignment(courseId, oauthToken, attendanceAssignment.getCanvasAssignmentId().toString());
+        } catch (IOException e) {
+            LOG.error("Error while getting assignment from canvas for section: " + attendanceAssignment.getAttendanceSection().getCanvasSectionId(), e);
+            throw new IOException("Could not connect to Canvas to get assignment");
+        }
+
+        if(!assignmentOptional.isPresent()) {
+            LOG.info("No Canvas assignment linked to section: " + attendanceAssignment.getAttendanceSection().getCanvasSectionId());
+            throw new CanvasOutOfSyncException("NO CANVAS ASSIGNMENT LINKED");
+        }
+    }
+
+    public void validateCanvasAssignment(AttendanceAssignment assignmentConfigurationFromSetup, Long courseId, AttendanceAssignment attendanceAssignment,
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Refactor this method to throw at most one checked exception instead of: java.io.IOException, edu.ksu.canvas.attendance.exception.CanvasOutOfSyncException [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonar.ome.ksu.edu/coding_rules#rule_key=squid%3AS1160)